src/event/cloudevent.ts

```diff
@@ -89,7 +89,7 @@ export class CloudEvent implements CloudEventV1 {
     this.schemaurl = properties.schemaurl as string;
     delete properties.schemaurl;
 
-    this.data = properties.data;
+    this.data = properties.data as T;
```

Comment:
this cast should be unnecessary

Reply:
The type checking in VSCode reports this if I remove the cast.

```
Type 'T | undefined' is not assignable to type 'T'.
  'T' could be instantiated with an arbitrary type which could be unrelated to 'T | undefined'.ts(2322)
(property) CloudEvent<T = undefined>.data: T = undefined
```